contributing/development/cpp_usage_guidelines.rst

```diff
@@ -53,18 +53,69 @@ as Godot provides its own data types (among other things).
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Span``               | ``std::span``            | Represents read-only access to a contiguous array without needing to copy any data. |
+|                        |                          | See `pull request description <https://github.com/godotengine/godot/pull/100293>`__ |
+|                        |                          | for details.                                                                        |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``HashSet``            | ``std::set``             | **Use this as the "default" set type.**                                             |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``RBSet``              | ``std::set``             | Uses a `red-black tree <https://en.wikipedia.org/wiki/Red-black_tree>`__            |
+|                        |                          | for faster access.                                                                  |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``HashMap``            | ``std::map``             | **Use this as the "default" map type.** Iteration is different from ``std::map``    |
+|                        |                          | (``HashMap`` is insertion-ordered, while ``std::map`` is key-ordered).              |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``AHashMap``           | ``std::map``             | Array-based implementation of a hash map.                                           |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``OAHashMap``          | ``std::unordered_map``   | Does not preserve insertion order. Generally faster than ``HashMap`` and            |
+|                        |                          | ``AHashMap``.                                                                       |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Dictionary`` 📜      | ``std::map``             | Keys and values can be of any Variant type. No static typing is imposed.            |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``TypedDictionary`` 📜 | ``std::map``             | Equivalent to ``Dictionary`` but with static typing for its keys and values.        |
```

Comment:
Same for `Dictionary`, as they are just `HashMap`, and again aren't ordered on key (unless sorted)